### PR TITLE
get single task function created

### PR DESCRIPTION
**Pull request by kevinyc-dri**
Please review our code to get a single task.

### CODE REVIEW

**Comment by kevinyc-dri** [comment]
Image 2021-04-22 at 11 35 46 AM

File: functions/index.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 const cors = require('cors')
-const { getTasks, createTask } = require('./src/tasks')
+const { getTasks, createTask, getSingleTask } = require('./src/tasks')
 
 const app = express()
 app.use(cors())
 
 app.get('/tasks', getTasks)
 app.post('/tasks', createTask)
+app.get('/singletask', getSingleTask)
```
Comment (toddalbert) [inline functions/index.js]: Need to pass a param

File: functions/src/tasks/index.js [hunk 2]
```diff
@@ -35,3 +35,11 @@ exports.createTask = (req, res) => {
     .then(() => this.getTasks(req, res))
     .catch(err => res.status(500).send('Error creating task: ' + err.message))
 }
+
+exports.getSingleTask = (req, res) => {
+  connectToFirestore()
+  const singleTask = req.params
+  db.collection('tasks').get(singleTask)
```
Comment (toddalbert) [inline functions/src/tasks/index.js]: Check firestore docs for how to get a single document

File: functions/src/tasks/index.js [hunk 3]
```diff
@@ -35,3 +35,11 @@ exports.createTask = (req, res) => {
...
+    .then(() => this.getTasks(req, res))
```
Comment (toddalbert) [inline functions/src/tasks/index.js]: Don't think you want to call this function

**Review by toddalbert — CHANGES_REQUESTED** [review]
See comments in file

File: functions/index.js (hunk 1 above)
Comment (kevinyc-dri) [reply functions/index.js]: `app.get('/tasks/:taskId', getSingleTask)`

File: functions/src/tasks/index.js (hunk 2 above)
Comment (kevinyc-dri) [reply functions/src/tasks/index.js]:
```exports.getSingleTask = (req, res) => {
  connectToFirestore()
  const { taskId } = req.params
  db.collection('tasks')
    .doc(taskId)
    .get()
    .then(singleTask => res.send(singleTask.data()))
}```

File: functions/src/tasks/index.js (hunk 3 above)
Comment (kevinyc-dri) [reply functions/src/tasks/index.js]:
```exports.getSingleTask = (req, res) => {
  connectToFirestore()
  const { taskId } = req.params
  db.collection('tasks')
    .doc(taskId)
    .get()
    .then(singleTask => res.send(singleTask.data()))
}```

Comment (toddalbert) [reply functions/src/tasks/index.js]: YES!

**Review by toddalbert — APPROVED** [review]
Awesome

**Pull request merged**